src/core/model_config.proto

```diff
@@ -764,6 +766,30 @@ message ModelParameter
   string string_value = 1;
 }
 
+//@@
+//@@.. cpp:var:: message ModelWarmUp
+//@@
+//@@   Settings for warming up the model instances.
+//@@
+message ModelWarmUp
+{
+  //@@  .. cpp:var:: InferRequestHeader request_header
+  //@@
+  //@@     The header message of the request sample.
```

Comment:
"The request meta-data to use for the warm-up inference request."
But I'm actually not sure we should do this. There is a lot of information in the request header that just doesn't apply here. Really all we need are the shapes for inputs that have variable-size dimensions (other than batch dimension). So can we just have a map from tensor name to dims (and even that is required only for inputs that need it). We should be able to generate the rest of the meta-data.